core/trino-main/src/main/java/io/trino/execution/buffer/PartitionedOutputBuffer.java

```diff
@@ -187,12 +191,13 @@ public void enqueue(int partitionNumber, List<Slice> pages)
         }
         List<SerializedPageReference> serializedPageReferences = references.build();
 
+        // reserve memory
+        memoryManager.updateMemoryUsage(bytesAdded);
+
         // update stats
         totalRowsAdded.addAndGet(rowCount);
         totalPagesAdded.addAndGet(serializedPageReferences.size());
-
-        // reserve memory
-        memoryManager.updateMemoryUsage(bytesAdded);
+        outputBufferUtilizationDigest.add(getUtilization());
```

Comment:
This should still account for time from last update. We want to know histogram of buffer utilization over entire duration of query

Reply:
Why do you think that we need to weight that value in time. This value is per query / task and it should be sufficient for us to have the distribution of utilzation of that. We sample it for every change of utilization.